lib/PSR/Chain.hs

```diff
@@ -40,22 +44,47 @@ getTxInSet tx =
         txBodyContent = C.getTxBodyContent txBody
      in Set.fromList $ map fst $ C.txIns txBodyContent
 
-queryInputUtxoMap ::
-    forall era.
-    C.LocalNodeConnectInfo ->
-    C.ChainPoint ->
+utxoMapQuery ::
     C.ShelleyBasedEra era ->
     [C.Tx era] ->
-    IO (Map C.TxIn (C.TxOut C.CtxUTxO era))
-queryInputUtxoMap conn cp era txList = do
+    C.LocalStateQueryExpr block point C.QueryInMode r IO (C.UTxO era)
+utxoMapQuery era txList = do
     let txInsSet = Set.unions $ getTxInSet <$> txList
-        query = C.queryUtxo era $ C.QueryUTxOByTxIn txInsSet
+    res <- C.queryUtxo era $ C.QueryUTxOByTxIn txInsSet
+    case res of
+        Left err -> throw $ QeUnsupportedNtcVersionError err
+        Right (Left err) -> throw $ QeEraMismatch err
```

Comment:
Not sure I'm happy with the exceptions, but it'll do for now. This will mean the app crashes as soon as one of these happens right?